sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/scanner/MappableField.java

```diff
@@ -134,13 +150,61 @@ private String createPropertyTypeDescription(Type type) {
       if (Map.class.isAssignableFrom(clazz)) {
         return "Map";
       }
+      if (clazz.isArray() || Collection.class.isAssignableFrom(clazz)) {
+        return "Array";
+      }
       return clazz.getSimpleName();
     }
     return type.getTypeName();
   }
 
   private Pattern createExpandPattern(String environmentVariableName) {
-    return Pattern.compile(
-        "^" + environmentVariableName.replace(MAP_KEY_PLACE_HOLDER_IN_CONTEXT_KEY, "(.*)") + "$");
+    var envWithMapKeyPattern =
+        environmentVariableName.replace(MAP_KEY_PLACE_HOLDER_IN_CONTEXT_KEY, "(.*)");
+    var envWithMapKeyAndArrayIndexPattern =
+        envWithMapKeyPattern.replace(ARRAY_INDEX_PLACE_HOLDER_IN_CONTEXT_KEY, "(\\d*)");
+    return Pattern.compile("^" + envWithMapKeyAndArrayIndexPattern + "$");
+  }
+
+  @Override
+  public int compareTo(@NotNull MappableField o) {
```

Comment:
I'd add some tests for compareTo/equals/hashCode.

Reply:
Yes definitely. One reason why this is still a draft.

Reply:
btw. `Comparable` is needed because, lists (specifically `ArrayNodes`) can only be updated in order. It does not allow to insert index 1 if no item at index 0 exists.